### PR TITLE
Update apidoc scripts

### PR DESCRIPTION
**Pull request by sgtcoolguy**
- Move apidoc dependencies to top-level package.json as devDependencies. (So instead of doing `cd apidoc; npm install; node docgen.js` you'd do `npm install; cd apidoc; node docgen.js`)
- Fix up ejs usage to work on ejs 2+.
- Remove old python files for apidoc.
- Update the validate.js to be more strict about required properties as our old validate.py was.
- Validate the docs as part of our build process.
- Fix a doc bug in Ti.UI.Clipboard where 3 properties didn't have a type specified (which the new validate.js found but validate.py did not!)

### CODE REVIEW

**Comment by sgtcoolguy** [comment]
cc @jawa9000 I'm not sure if there's other build scripts to update related to this? Basically the big change is that there's no separate package.json inside the apidoc folder with this change, and the `npm install` would run at the root folder of the repo, not inside apidoc.

File: apidoc/validate.py [hunk 1]
```diff
@@ -1,623 +0,0 @@
-#!/usr/bin/env python
```
Comment (hansemannn) [inline apidoc/validate.py]: We should not remove that particular script, yet. It has some syntax-validations that the node-one does not currently have. I usually use both to scan changes, so I could setup a table of missing checks if necessary.

Comment (sgtcoolguy) [inline apidoc/validate.py]: I'd suggest we simply fix the validate.js file to include the additional checks. I don't think it's good for us to have two separate and unequal validation scripts in entirely different language/toolsets.

Comment (sgtcoolguy) [inline apidoc/validate.py]: Do you happen to know off-hand what the additional checks are? Or do we need to compare the two file and try and figure it out from there?

Comment (hansemannn) [inline apidoc/validate.py]: I would just change some docs with typical errors and check both scripts, no big thing to do. I can give you at least a list of differences by tomorrow if it helps. It's only about small things like disallowed the `description` key in function parameters as I remember. The actual docs build script would ignore them anyway.

Comment (sgtcoolguy) [inline apidoc/validate.py]: I'm taking a look now. Looks like a few key differences are that the python script:
- enforced some properties were required on some structures (typically: name, summary)
- enforced a couple properties were valid markdown (not just strings)
- enforce properties whose name were all-caps had permission explicitly set to 'read-only'
- complained the 'notes' fields were '"no longer valid"

Related: should our build process be running the doc validation at some point? Sounds like maybe we should be doing this in an automated way to reject PRs with invalid doc changes...

Comment (hansemannn) [inline apidoc/validate.py]: That would be **awesome**, really.

**Comment by hansemannn** [comment]
Are there more files we clean up in that directory? For example the Readme.md, which includes notes regarding a "tdoc" format, which isn't used since forever.

**Comment by sgtcoolguy** [comment]
I removed the README.md, not sure what else could go. Maybe @jawa9000 has a better idea.

I did see that before (and after) my changes, docgen was broken for `modulehtml` format.

**Comment by sgtcoolguy** [comment]
Since this is a dev/build related change and doesn't affect SDK functionality, I'm gonna merge it.

**Pull request merged**